bazel/p4_library.bzl

```diff
@@ -2,26 +2,62 @@
 
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
 
-def _p4_library_impl(ctx):
-    p4c = ctx.executable._p4c
+def _extract_common_p4c_args(ctx):
+    """Extract input files and common arguments for p4c build rules."""
     p4file = ctx.file.src
     p4deps = ctx.files._p4include + ctx.files.deps
-    target = ctx.attr.target
     args = [
         p4file.path,
         "--std",
         ctx.attr.std,
-        "--target",
-        (target if target else "bmv2"),
         "--arch",
         ctx.attr.arch,
     ]
-    if ctx.attr.extra_args:
-        args.append(ctx.attr.extra_args)
 
     include_dirs = {d.dirname: 0 for d in p4deps}  # Use dict to express set.
     include_dirs["."] = 0  # Enable include paths relative to workspace root.
     args += [("-I" + dir) for dir in include_dirs.keys()]
+    inputs = p4deps + [p4file]
+
+    return (args, inputs)
+
+def _run_p4c_with_cc(ctx, p4c, command, **run_shell_kwargs):
```

Comment:
How about simplifying this to "_run_shell_cmd_with_p4c(ctx, command)"? In particular:
- The function can assert that `ctx.executable._p4c` is defined.
- No need to mention cc, that's arguably an implementation detail.
- We can add `run_shell_kwargs` if we need it in the future, currently it is unused.

Reply:
The build rule implementations pass their inputs, outputs, and progress messages via `run_shell_kwargs`. I can add them as explicit parameters instead, if that makes sense.

I simplifed it with other changes. I used `if` and `fail` because Starlark does not have assertions (outside unit tests, which do not fail immediately) as far as I am aware of.